From f6ef8dcfdbe4ee3252a5b24f19d5a39ce56c8455 Mon Sep 17 00:00:00 2001
From: akgupta <akgupta@maxlinear.com>
Date: Wed, 26 Jun 2024 14:06:49 +0530
Subject: [PATCH] WLANRTSYS-82491 [WAV700][MLD] Beacon protection is not
 enabled by default in Latest builds.

Issue: Beacon protection is not enabled by default for 11be mode in hostap-ng.
Fix: IEEE P802.11be/D3.0, 4.3.16a Extremely high throughput (EHT) STA
requires beacon protection support for an EHT AP, therefore beacon
protection is enabled if IEEE 802.11be is set for BSS.
---
 src/ap/ap_config.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/ap/ap_config.c b/src/ap/ap_config.c
index aca50cb28..ecdcd7c9b 100644
--- a/src/ap/ap_config.c
+++ b/src/ap/ap_config.c
@@ -1621,6 +1621,12 @@ static int hostapd_config_check_bss(struct hostapd_bss_config *bss,
 		wpa_printf(MSG_INFO,
 			   "Disabling IEEE 802.11be as IEEE 802.11ax is disabled for this BSS");
 	}
+
+	if (full_config && conf->ieee80211be && !bss->disable_11be && !bss->beacon_prot) {
+		bss->beacon_prot = 1;
+		wpa_printf(MSG_INFO,
+			   "Enabling beacon protection as IEEE 802.11be is enabled for this BSS");
+	}
 #endif /* CONFIG_IEEE80211BE */
 
 	if (full_config && bss->ignore_broadcast_ssid && conf->mbssid) {
-- 
2.43.0

